waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -117,10 +117,11 @@ genPublicDir spec = do
     publicIndexHtmlFd :
     ( let tmplData = object ["appName" .= (fst (getApp spec) :: String)]
           processPublicTmpl path = C.mkTmplFdWithData (C.asTmplFile $ [reldir|public|] </> path) tmplData
-       in processPublicTmpl
-            <$> [ [relfile|manifest.json|]
-                ]
-    )
+       in processPublicTmpl [relfile|manifest.json|]
+    ) :
+      [C.mkTmplFd (C.asTmplFile [relfile|public/images/btn_google_signin_dark_normal_web@2x.png|]) | (AS.App.Auth.isGoogleAuthEnabled <$> maybeAuth) == Just True]
+  where
+    maybeAuth = AS.App.auth $ snd $ getApp spec
```

Comment:
This got a bit difficult to read IMO, I think we'd benefit from extracting each list element into a named variable under the where clause.

Reply:
Yeah I can try to clean up the existing stuff and separate them all out 👍🏻